### PR TITLE
Binary for scheduler

### LINKED ISSUE #147: Create binary/image for scheduler
To deploy the scheduler, we need to build it into a binary.

Open questions:

- Does it make sense to ship one binary/image that provides every component, and the desired component is specified on the command line?

### PR DESCRIPTION
**Pull request by paulgb**
This PR:
- Does some light refactoring to the `drone` crate in preparation for adding a `controller` binary.
- Moves some common cli-related functionality out of `drone` and into `core`
- Standardizes the handling of functions that never return
- Implements a config parsing / planning / running pipeline mirroring the implementation in `drone`
- Stubs out DNS serving from the perspective of that pipeline.

This PR will close #147

### CODE REVIEW

**Review by pretentious7 — APPROVED** [review]
lgtm

File: core/src/cli.rs [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+            .try_parsing(true)
+            .list_separator(",")
+            .with_list_parse_key("nats.hosts"),
+    );
+    let config: C = config_builder.build()?.try_deserialize()?;
+
+    if cli_args.dump_config {
+        println!("{}", serde_json::to_string_pretty(&config)?);
+        // return Ok(());
+        todo!("exit gracefully");
```
Comment (rolyatmax) [inline core/src/cli.rs:34]: TIL about `todo!`

File: controller/src/run.rs [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    tracing_handle.attach_nats(nats.clone())?;
+
+    let mut futs: Vec<Pin<Box<dyn Future<Output = NeverResult>>>> = vec![];
+
+    if let Some(_) = scheduler_plan {
+        futs.push(Box::pin(run_scheduler(nats.clone())))
+    }
+
+    if let Some(_) = dns_plan {
+        todo!("DNS server not let implemented.")
```
Comment (rolyatmax) [inline controller/src/run.rs:32]: minor: typo

Comment (paulgb) [reply controller/src/run.rs]: fixed

**Pull request merged**